lib/lorekeeper/json_logger.rb

```diff
@@ -73,7 +73,7 @@ def exception(exception, custom_message = nil, custom_data = nil, custom_level =
       log_level = METHOD_SEVERITY_MAP[param_level] || ERROR
 
       if exception.is_a?(Exception)
-        backtrace = exception.backtrace || []
+        backtrace = clean_backtrace!(exception.backtrace || [])
```

Comment:
While I understand the rationale behind modifying in place I think it goes against the principle of least surprise. When I call `logger.exception(e)` I don't expect the logger to modify the exception object I might want to reuse somewhere else (for instance by re-raising).

Also, if you stick to the current approach, I'd do
```ruby
backtrace = exception.backtrace || []
clean_backtrace!(backtrace)
```
because `!` method calls shouldn't rely on the return value, which can be surprising. Remember the `gsub!` trap returning the value if changes were made but returning `nil` if there were no changes.

Reply:
You are perfectly right. For instance someone could think of freezing the exception object, why not? And this would cause an exception.
Updated the code to not modify the object.